### PR TITLE
VC and SA Mobile modes update

### PR DESCRIPTION
**Pull request by XMDS**
1.CLEO Android support(section)
-- Add VCMobile mode directory.
-- SAMobile mode data file organization.
See: https://github.com/sannybuilder/dev/issues/118
Sanny catalog of VCMobile mode should be prepared now
https://github.com/sannybuilder/dev/issues/6

2.Data file organization in PC VC mode

3.add opcodes-restoration-project
see: https://github.com/cleolibrary/opcodes-restoration-project

### CODE REVIEW

**Comment by XMDS** [comment]
[new_vc_monile](https://github.com/XMDS/GTA.III.VC.SA.CDK/releases/download/untagged-ee70a616d8af24393e11/vc_mobile.zip)
This is the sanny tool data directory of the new VC mobile mode, which I modified based on the release of OrionSR. 
@x87 You can review it and try to create SB's VC mobile mode in 3.7.0 (I think it is feasible). (The configuration of modes.xml does not support the current version, so I said you need to support this mode for the SB tool). In addition, the zip file also contains two different sizes of mobile version of VC game icons, which you can use as SB icons in VC mobile mode.

**Comment by XMDS** [comment]
@x87 I plan to submit the data catalog of VC Mobile mode, do you think it is feasible?

**Comment by x87** [comment]
@XMDS thank you. I'm curious how different VCM opcodes are from VC PC ones? Can we just use VCSCM.INI and opcodes.txt from the Vice City mode by extending vc mode?

**Comment by XMDS** [comment]
I updated the name definition of the new OP of SA MOBILE.  They are now correctly defined.

File: sa_mobile/SASCM.ini [hunk 1]
```diff
@@ -1968,70 +1968,90 @@ DATE=2020-11-11
...
+0a89=3,set_widget %1d% value2 %2d% %3d% ; float 
+0a8a=8,set_widget %1d% info %2d% %3d% %4d% %5d% %6d% %7d% %8g%
+0a8b=7,set_widget %1d% info2 %2d% %3d% %4d% %5d% %6g% %7g%
+0a8c=0,load_all_requested_models ; No time to pause. load faster than 038B
 0a8d=1,  is_hid_released %1d%
-0A8E=1,award_achievement %1d%
-0A8F=0,
-0A90=1,  %1d%
\ No newline at end of file
+0A8E=1,NOP %1d%
```
Comment (x87) [inline sa_mobile/SASCM.ini:2038]: Are you sure this is a no-operation? I remember there was a discussion somewhere on GTAForums about this opcode and somehow it was figured this grants new achievements to the player. I haven't tested it myself though

Comment (x87) [reply sa_mobile/SASCM.ini:2038]: https://gtaforums.com/topic/929701-new-opcodes-in-the-remastered-scm-of-san-andreas/?do=findComment&comment=1070954748

**Comment by XMDS** [comment]
@x87  I think that in addition to the 3 mobile mode created in a future version, my submissions are currently completed, you can review and merge them finally

File: vc_mobile/constants.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+false=0
```
Comment (x87) [inline vc_mobile/constants.txt:1]: Note to self: this file needs `const..end`  as described in Breaking Changes section
https://github.com/sannybuilder/dev/releases/tag/v3.7.0

File: sa_mobile/SASCM.ini (hunk 1 above)
Comment (x87) [reply sa_mobile/SASCM.ini:2038]: @XMDS ☝️

File: modes.xml [hunk 3]
```diff
@@ -35,6 +35,20 @@
...
+    <mode extends="vc" id="vc_mobile" title="VC Mobile" game="vc" type="">
+        <data>@sb:\data\vc_mobile\</data>
+        <opcodes>@sb:\data\vc_mobile\VCSCM.ini</opcodes>
+        <examples>@sb:\data\vc_mobile\opcodes.txt</examples>
+        <classes>@sb:\data\vc_mobile\classes.db</classes>  
+        <variables>@sb:\data\vc_mobile\CustomVariables.ini</variables>
+        <keywords>@sb:\data\vc_mobile\keywords.txt</keywords>
+        <missions>@sb:\data\vc_mobile\missions.txt</missions>        
+        <constants>@sb:\data\vc_mobile\constants.txt</constants>
+        <templates>@sb:\data\vc_mobile\templates.txt</templates>
```
Comment (x87) [inline modes.xml:47]: @XMDS Are those files all different in VC PC and VC Mobile ? Can we re-use something from VC PC mode and remove duplicated files?

**Comment by XMDS** [comment]
@x87 Except for the necessary files such as OP.txt and scm.txt for keyword expansion, I have deleted the duplicate files of other VC and VCM and submitted them

**Pull request merged**